src/interpolate.jl

```diff
@@ -40,6 +40,113 @@ function cell2node(cell_centered_data::AbstractArray{Float64})
+  nodes_in, _ = gauss_lobatto_nodes_weights(n_nodes_in)
+
+  # New unstructured data has one dimension less.
+  # The redundant element ids are removed later.
+  new_unstructured_data = similar(unstructured_data[1, ..])
+
+  # Declare new empty arrays to fill in new coordinates and levels
+  new_coordinates = Array{Float64}(undef, 2, 0)
+  new_levels = Array{eltype(levels)}(undef, 0)
+
+  # Counter for new element ids
+  new_id = 0
+
+  # Save vandermonde matrices in a Dict to prevent redundant generation
+  vandermonde_to_2d = Dict()
+
+  # Limits of domain in slice_axis dimension
+  lower_limit = center_level_0[slice_axis_dimension] - length_level_0 / 2
+  upper_limit = center_level_0[slice_axis_dimension] + length_level_0 / 2
+
+  if slice_axis_intersect < lower_limit || slice_axis_intersect > upper_limit
+    error("slice_axis_intersect outside of domain")
+  end
+
+  for element_id in 1:n_elements
+    # Distance from center to border of this element (half the length)
+    element_length = length_level_0 / 2^levels[element_id]
+    first_coordinate = coordinates[:, element_id] .- element_length / 2
+    last_coordinate = coordinates[:, element_id] .+ element_length / 2
+
+    # Check if slice plane and current element intersect
+    # The upper limit check is needed because of the > in the first check
+    if (first_coordinate[slice_axis_dimension] <= slice_axis_intersect &&
+          last_coordinate[slice_axis_dimension] > slice_axis_intersect) ||
+        (slice_axis_intersect == upper_limit &&
+          last_coordinate[slice_axis_dimension] == upper_limit)
+      # This element is of interest
+      new_id += 1
+
+      # Add element to new coordinates and levels
+      new_coordinates = hcat(new_coordinates, coordinates[other_dimensions, element_id])
+      push!(new_levels, levels[element_id])
+
+      # Construct vandermonde matrix (or load from Dict if possible)
+      normalized_intersect =
+          (slice_axis_intersect - first_coordinate[slice_axis_dimension]) /
+          element_length * 2 - 1
+
+      if haskey(vandermonde_to_2d, normalized_intersect)
+        vandermonde = vandermonde_to_2d[normalized_intersect]
+      else
+        # Generate vandermonde matrix to interpolate values at nodes_in to one value
+        vandermonde = polynomial_interpolation_matrix(nodes_in, [normalized_intersect])
+        vandermonde_to_2d[normalized_intersect] = vandermonde
+      end
+
+      # 1D interpolation to specified slice plane
+      for i in 1:n_nodes_in
+        for ii in 1:n_nodes_in
+          for v in 1:n_variables
+            if slice_axis == :x
+              data = unstructured_data[:, i, ii, element_id, v]
+            elseif slice_axis == :y
+              data = unstructured_data[i, :, ii, element_id, v]
+            elseif slice_axis == :z
+              data = unstructured_data[i, ii, :, element_id, v]
+            end
+            value = vandermonde * data
```

Comment:
I probably should have used `interpolate_nodes` from `interpolation`. Will have a look later today.